### PR TITLE
Check digit errors

### LINKED ISSUE #1427: Check-digit endpoint: Return sensible error for invalid input
Currently, if the input loan ID contains any non-alphanumeric characters (such as `!#$%^`), it returns `500 Internal Server Error` and no helpful information.

Instead, use response code `400` and return a useful error message, e.g. "LEI + Loan/Application ID may only contain alphanumeric characters".

### PR DESCRIPTION
**Pull request by jmarin**
Closes #1427

### CODE REVIEW

File: api/src/test/scala/hmda/api/http/public/ULIHttpApiSpec.scala [hunk 1]
```diff
@@ -10,8 +10,11 @@ import hmda.api.protocol.public.ULIProtocol
 import hmda.model.fi.lar.LarGenerators
```
Comment (schbetsy) [inline api/src/test/scala/hmda/api/http/public/ULIHttpApiSpec.scala]: `LarGenerators` aren't used in this spec.

File: validation/src/main/scala/hmda/validation/engine/lar/ULIValidationErrorMessages.scala [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package hmda.validation.engine.lar
+
+object ULIValidationErrorMessages {
+
+  final val invalidLoanIdLengthMessage = "Loan ID is not between 21 and 43 characters long"
```
[the remark this answers is not in the document]
Comment (jmarin) [reply validation/src/main/scala/hmda/validation/engine/lar/ULIValidationErrorMessages.scala]: I prefer the original message. Leaving as is

File: api/src/test/scala/hmda/api/http/public/ULIHttpApiSpec.scala [hunk 3]
```diff
@@ -82,6 +104,25 @@ class ULIHttpApiSpec extends WordSpec with MustMatchers with BeforeAndAfterAll
...
+        response.path mustBe Path("/uli/validate")
+      }
+      Post("/uli/validate", nonAlphaNumericCheck) ~> uliHttpRoutes ~> check {
+        status mustBe StatusCodes.BadRequest
+        val response = responseAs[ErrorResponse]
+        response.path mustBe Path("/uli/validate")
+        response.message mustBe nonAlphanumericULIMessage
+      }
     }
     "validate a file of ULIs and return csv" in {
```
[the remark this answers is not in the document]
Comment (jmarin) [reply api/src/test/scala/hmda/api/http/public/ULIHttpApiSpec.scala]: There was no mention of how batch should behave in this ticket. The behavior [implemented](https://github.com/jmarin/hmda-platform/blob/check-digit-errors/api/src/main/scala/hmda/api/http/public/ULIHttpApi.scala#L139) in this PR filters out lines that produce errors due to not being alphanumeric or with invalid length, and they won't be included in the response (json or csv). I've added bad data to the test to prove this. If we end up needing a different behavior I suggest creating a different ticket

[the remark this answers is not in the document]
Comment (schbetsy) [reply api/src/test/scala/hmda/api/http/public/ULIHttpApiSpec.scala]: Okay. I've created https://github.com/cfpb/hmda-platform/issues/1438 for updating the batch behavior.

**Pull request merged**